src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -105,9 +132,65 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     return true;
   }
 
+  /**
+   * vectorize replacementDocument as needed, only when document is found will be used by
+   * findOneAndReplace
+   *
+   * @param dataVectorizer
+   */
+  public Uni<Boolean> vectorizeTheReplacementDocument(DataVectorizer dataVectorizer) {
+    return dataVectorizer.vectorize(List.of(replaceDocument));
+  }
+
+  /**
+   * Check if there is any $vectorize diff If there are docs found to update or doc to replace, then
+   * this is a necessary condition to proceed vectorization
+   *
+   * @param foundDocs
+   */
+  public boolean hasVectorizeDiff(List<ReadDocument> foundDocs) {
+    String vectorizeTextUpdate = null;
+    if (updateType().equals(DocumentUpdater.UpdateType.UPDATE)) {
+      // extract $vectorize if updateClause set operator has it
+      final ObjectNode setNode = updateClause.updateOperationDefs().get(UpdateOperator.SET);
+      if (setNode != null) {
+        final JsonNode updateClauseVectorizeTextJsonNode =
+            setNode.get(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
+        if (updateClauseVectorizeTextJsonNode != null) {
+          vectorizeTextUpdate = updateClauseVectorizeTextJsonNode.asText();
+        }
+      }
+    } else if (updateType().equals(DocumentUpdater.UpdateType.REPLACE)) {
+      // extract $vectorize if replaceDocument has it
+      final JsonNode replaceDocumentVectorizeTextJsonNode =
+          replaceDocument.get(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
+      if (replaceDocumentVectorizeTextJsonNode != null) {
+        vectorizeTextUpdate = replaceDocumentVectorizeTextJsonNode.asText();
+      }
+    }
+
+    // if there is no $vectorize to update or replace, then no diff.
+    if (vectorizeTextUpdate == null) {
+      return false;
+    }
+
+    // iterate foundDocs, see if there is any diff for $vectorize
+    for (ReadDocument foundDoc : foundDocs) {
+      final JsonNode foundDocVectorizeTextJsonNode =
+          foundDoc.document().get(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
+      if (foundDocVectorizeTextJsonNode != null) {
+        if (!foundDocVectorizeTextJsonNode.asText().equals(vectorizeTextUpdate)) {
+          // There is a diff
+          return true;
```

Comment:
what if there are 10 docs, and only the first was changed ? 

Also want to check if we can just compare the docs before and after, that feels a safer and easier that extracting something from the update clause